File: idea-plugin/src/main/java/org/ifinalframework/jetbrains/plugins/aio/issue/IssueOpener.java
package org.ifinalframework.jetbrains.plugins.aio.issue;


import com.intellij.psi.PsiElement;

/**
 * IssueOpener
 *
 * @author iimik
 * @since 0.0.1
 **/
public interface IssueOpener {

    boolean isSupported(PsiElement element);

    void open(PsiElement element);

}

File: idea-plugin/src/main/kotlin/org/ifinalframework/plugins/jetbrains/aio/issue/IssueOpener.kt
package org.ifinalframework.plugins.jetbrains.aio.issue


/**
 * IssueOpener
 *
 * @author iimik
 * @since 0.0.2
 **/
interface IssueOpener {

    fun supported(issueType: IssueType): Boolean

    fun open(issue: Issue)
}